### PR TITLE
feat(cjs-module-analyzer): Add package for lexical analysis of CommonJS modules

### PR DESCRIPTION
**Pull request by kriskowal**
In preparation for adding CommonJS support back to Endo, this adds a package forked off https://github.com/guybedford/cjs-module-lexer that allows us to statically analyze `imports`, `exports`, and `reexports` (`export * from _`).

The fork is required because import graph analysis is not required for Node.js upstream, since its module loader is still synchronous. I’ve discussed with @guybedford the possibility of collaboration, and it seems the best we can do for now is attempt to make the fork as similar to the original as possible to make tracking easy. Changes should not be frequent.

Part two of #501 #655

### CODE REVIEW

**Comment by kriskowal** [comment]
The commit history reflects the steps I took to make the original source play well with our lint and TypeScript checks and what changes I made locally. These I intend to preserve as a transcript to guide future maintenance.

File: package.json [hunk 1]
```diff
@@ -4,6 +4,7 @@
   "useWorkspaces": true,
   "workspaces": [
     "packages/base64",
+    "packages/cjs-module-analyzer",
```
Comment (michaelfig) [inline package.json:7]: Again, why not `"packages/*"`?

**Review by michaelfig — APPROVED** [review]
I did not actually read all the lexer.  I am committing to be able to support things if they break, given that the test suite is currently passing.

The Endoisms are fine.

File: package.json (hunk 1 above)
Comment (kriskowal) [reply package.json:7]: I changed this to `packages/*` in Agoric SDK one time and someone reverted it back. I don’t recall what it broke, but I believe some tool takes this list literally.

**Pull request merged**